fvm/evm/stdlib/contract.cdc

```diff
@@ -149,6 +149,53 @@ contract EVM {
             )
             emit FLOWTokensDeposited(addressBytes: self.bytes, amount: amount)
         }
+
+        /// Serializes the address to a hex string without the 0x prefix
+        /// Future implementations should pass data to InternalEVM for native serialization
+        access(all)
+        view fun toString(): String {
+            let addressBytes: [UInt8] = [
+                self.bytes[0], self.bytes[1], self.bytes[2], self.bytes[3],
+                self.bytes[4], self.bytes[5], self.bytes[6], self.bytes[7],
+                self.bytes[8], self.bytes[9], self.bytes[10], self.bytes[11],
+                self.bytes[12], self.bytes[13], self.bytes[14], self.bytes[15],
+                self.bytes[16], self.bytes[17], self.bytes[18], self.bytes[19]
+            ]
+            return String.encodeHex(addressBytes)
+        }
+
+        /// Compares the address with another address
+        access(all)
+        view fun equals(_ other: EVMAddress): Bool {
+            return self.bytes == other.bytes
+        }
+    }
+
+    /// Converts a hex string to an EVM address if the string is a valid hex string
+    /// Future implementations should pass data to InternalEVM for native deserialization
+    access(all)
+    fun addressFromString(_ asHex: String): EVMAddress? {
+        pre {
+            asHex.length == 40 || asHex.length == 42: "Invalid hex string length"
+        }
+        // Remove the 0x prefix if it exists
+        let sanitized = (asHex[1] == "x" ? asHex.split(separator: "x")[1] : asHex).toLower()
+        if sanitized.length != 40 {
+            return nil
+        }
+        // Decode the hex string
+        var addressBytes: [UInt8] = asHex.decodeHex()
+        if addressBytes.length != 20 {
+            return nil
+        }
```

Comment:
The function returns an optional value, so seems to indicate it will return nil for invalid inputs.
However, it sometimes returns nil, sometimes panics due to the pre-conditions. Given that `decodeHex` aborts if the input is invalid, maybe just consistently panic in all cases, or if this needs to return an optional to handle invalid cases, also validate `asHex` only has valid hex-chars.

Maybe just this?

```cadence
fun addressFromString(_ asHex: String): [UInt8;20] {
    var withoutPrefix = asHex
    // Strip the 0x prefix, if any
    if asHex.length >= 2 && asHex[0] == "0" && asHex[1] == "x" {
        withoutPrefix = asHex.slice(from: 2, upTo: asHex.length)
    }
    return withoutPrefix.decodeHex().toConstantSized<[UInt8;20]>()!
}
```

Reply:
Good point. I'd love if I could run decodeHex in a try block and return nil in a catch, but since that's not possible I'd prefer to revert on invalid inputs for the sake of consistency.

> `fun addressFromString(_ asHex: String): [UInt8;20] {`

Are you suggesting we change the return type as well or was this just for illustration?